### PR TITLE
new-app: Use the right image stream for the build from

### LINKED ISSUE #1996: osc new-app is not using the right image stream for the build from
Scenario:

1. Create an image stream in namespace openshift called "ruby", set dockerImageRepository to "openshift/ruby-20-centos7"
2. Run `osc new-app openshift/ruby~<url>`

Build config has:
```
    strategy:
      stiStrategy:
        builderImage: openshift/ruby-20-centos7
        clean: true
        image: openshift/ruby-20-centos7
```

which is wrong, it should be "from" -> "openshift/ruby:latest".

### PR DESCRIPTION
**Pull request by kargakis**
Fixes #1996

@smarterclayton @bparees PTAL

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

File: pkg/generate/app/app.go [hunk 1]
```diff
@@ -144,23 +144,28 @@ type BuildStrategyRef struct {
...
+		}, s.Base.BuildTriggers()
 	}
 
 	return &buildapi.BuildStrategy{
 		Type: buildapi.STIBuildStrategyType,
 		STIStrategy: &buildapi.STIBuildStrategy{
 			From: &kapi.ObjectReference{
-				Kind: "DockerImage",
-				Name: s.Base.String(),
+				Kind: "ImageStreamTag",
```
Comment (smarterclayton) [inline pkg/generate/app/app.go]: You also need to set the namespace if the image is from a different namespace.  So ImageRef needs to be able to tell you where it comes from - I would suggest having ImageRef generate an ObjectReference now (ObjectReference() *kapi.ObjectReference) and always return that, now that ben's changes are in.

**Comment by kargakis** [comment]
al_green-209048

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**